src/modules/expression/expr.rs

```diff
@@ -101,6 +101,14 @@ impl Typed for Expr {
 }
 
 impl Expr {
+    pub fn get_integer_value(&self) -> Option<isize> {
```

Comment:
This function is intended to return an integer for compile-time arithmetic, but only when given integer literals like `42` or `-42`. We're not worrying about float literals like `123.45` because this will only be called in contexts where an integer is required.

Reply:
I was tempted to put this functionality into the `TranslateModule` trait, but held off for now because it's fairly niche at the moment.  However, I think doing compile time arithmetic where possible cannot be a bad idea; and we may decide to refactor if it becomes more widely used.